### PR TITLE
[Profiler] Clean up profiler includes.

### PR DESCRIPTION
**Pull request by robieta**
Summary: I've hit a lot of build issues in D32671972, and I've come to realize that a lot of it boils down to header hygene. `function.h` includes `profiler.h` *solely* to transitively include `record_function.h` which winds up leaking the profiler symbols. Moreover several files are relying on transitive includes to get access to `getTime`. As long as I have to touch all the places that use `getTime`, I may as well also move them to the new namespace.

Test Plan: Unit tests and CI.

Differential Revision: D32865907

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

### CODE REVIEW

**Review by albanD — APPROVED** [review]
SGTM

File: torch/csrc/jit/mobile/profiler_edge.cpp [hunk 1]
```diff
@@ -18,49 +18,52 @@ KinetoEdgeCPUProfiler::KinetoEdgeCPUProfiler(
...
-          // just construct a temporary vector using one string element
-          if (this->m_.hasDebugHandles()) {
-            e.moduleHierarchy(std::vector<std::string>(
-                {this->m_.getModuleHierarchy(e.debugHandle())}));
-          } else {
-            e.moduleHierarchy(std::vector<std::string>({no_debug_info}));
+    auto post_processing =
+        [this, with_stack, with_modules](
+            std::vector<torch::autograd::profiler::KinetoEvent>& events) {
+          std::string no_debug_info(
```
Comment (aaronenyeshi) [inline torch/csrc/jit/mobile/profiler_edge.cpp:34]: nit: should indent be 2 spaces from line 31?

**Review by aaronenyeshi — APPROVED** [review]
LGTM, thank you for transferring some of the functions out of autograd namespace. Small nit, but won't block this on that.

**Pull request closed without merging**